src/hb-ot-layout-gsubgpos.hh

```diff
@@ -2002,6 +2002,77 @@ struct ChainRule
 					      lookup.arrayZ, lookup_context));
   }
 
+  void serialize (hb_serialize_context_t *c,
+                  const hb_map_t *backtrack_map,
+                  const hb_map_t *input_map = nullptr,
+                  const hb_map_t *lookahead_map = nullptr) const
+  {
+    const hb_map_t *mapping = backtrack_map;
+    if (unlikely (!c->allocate_size<ArrayOf<HBUINT16>> (backtrack.get_size ()))) return;
+    c->copy (backtrack.len);
```

Comment:
Make a helper method for this block of code, since it's identical to the for each of backtrack, input, and lookahead.